commands/show_test.go

```diff
@@ -59,48 +71,69 @@ func TestBlockDaemon(t *testing.T) {
-	).Start()
-	defer minerDaemon.ShutdownSuccess()
+	fastenvOpts := fast.EnvironmentOpts{
+		//InitOpts: []fast.ProcessInitOption{fast.POAutoSealIntervalSeconds(1)},
+		//DaemonOpts: []fast.ProcessDaemonOption{fast.POBlockTime(time.Second)},
+	}
 
-	clientDaemon := th.NewDaemon(t,
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-		th.DefaultAddress(fixtures.TestAddresses[1]),
-	).Start()
-	defer clientDaemon.ShutdownSuccess()
+	ctx, env := fastesting.NewTestEnvironment(context.Background(), t, fastenvOpts)
 
-	minerDaemon.RunSuccess("mining", "start")
-	minerDaemon.UpdatePeerID()
+	clientNode := env.GenesisMiner
+	require.NoError(t, clientNode.MiningStart(ctx))
 
-	minerDaemon.ConnectSuccess(clientDaemon)
+	minerNode := env.RequireNewNodeWithFunds(1000)
+	defer func() {
+		require.NoError(t, env.Teardown(ctx))
+	}()
 
-	addAskCid := minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
+	// Connect the clientNode and the minerNode
+	err := series.Connect(ctx, clientNode, minerNode)
+	require.NoError(t, err)
 
-	t.Run("when deal exists shows deal info", func(t *testing.T) {
-		clientDaemon.WaitForMessageRequireSuccess(addAskCid)
-		dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
+	// Create a minerNode
+	collateral := big.NewInt(500)           // FIL
+	price := big.NewFloat(0.000000001)      // price per byte/block
+	expiry := big.NewInt(24 * 60 * 60 / 30) // ~24 hours
 
-		proposeDealOutput := clientDaemon.RunSuccess("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStdoutTrimNewlines()
+	ask, err := series.CreateStorageMinerWithAsk(ctx, minerNode, collateral, price, expiry)
+	require.NoError(t, err)
 
-		splitOnSpace := strings.Split(proposeDealOutput, " ")
+	// Create some data that is the full sector size and make it autoseal asap
+	var data bytes.Buffer
 
-		dealCid := splitOnSpace[len(splitOnSpace)-1]
+	maxBytesi64 := int64(getMaxUserBytesPerStagedSector())
+	dataReader := io.LimitReader(rand.Reader, maxBytesi64)
+	dataReader = io.TeeReader(dataReader, &data)
+	_, deal, err := series.ImportAndStore(ctx, clientNode, ask, files.NewReaderFile(dataReader))
+	require.NoError(t, err)
 
-		showdeal := minerDaemon.RunSuccess("show", "deal", dealCid).ReadStdoutTrimNewlines()
-		assert.Contains(t, showdeal, fmt.Sprintf("CID: %s", dealCid))
-		assert.Contains(t, showdeal, fmt.Sprintf("Miner: %s", fixtures.TestMiners[0]))
-		assert.Contains(t, showdeal, "Duration: 5 blocks")
-		assert.Contains(t, showdeal, "State: staged")
-		assert.Contains(t, showdeal, "Size: 12 bytes")
-		assert.Contains(t, showdeal, "Total Price: 1200 FIL")
+	t.Run("showDeal outputs correct information", func(t *testing.T) {
+		showDeal, err := clientNode.ShowDeal(ctx, deal)
+		require.NoError(t, err)
+
+		assert.Equal(t, ask.Miner.String(), showDeal.Miner.String())
+		assert.Equal(t, "accepted", showDeal.Response.State.String())
```

Comment:
These both can be checked against there value and don't need to be converted to strings first.